### PR TITLE
Adds Display Name Argument to Techweb Logging

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl:Cobby
code: Adds Stat Tracking to Techwebs by node and order.
/:cl:

Hopefully this works.

Purpose: Seeing what is researched most/least frequently and if people beeline for certain tech allows for more reactive balancing.

### CODE REVIEW

**Review by Jordie0608 — COMMENTED** [review]
?
This is already tracked and implicitly ordered by insertion and time at https://github.com/tgstation/tgstation/blob/master/code/modules/research/rdconsole.dm#L156

**Comment by Jordie0608** [comment]
If you want you could add `display_name` to the above, the id isn't always indicative of the tech name.

**Comment by ExcessiveUseOfCobblestone** [comment]
huh I was trying to look it up on Atlanta's site and didn't find it so I assumed it didn't exist.

I'll do that though.

File: code/modules/research/rdconsole.dm [hunk 1]
```diff
@@ -153,7 +153,7 @@ doesn't have toxins access.
 	if(stored_research.research_points >= price)
 		investigate_log("[key_name(user)] researched [id]([price]) on techweb id [stored_research.id].", INVESTIGATE_RESEARCH)
 		if(stored_research == SSresearch.science_tech)
-			SSblackbox.record_feedback("associative", "science_techweb_unlock", 1, list("id" = "[id]", "price" = "[price]", "time" = "[SQLtime()]"))
+			SSblackbox.record_feedback("associative", "science_techweb_unlock", 1, list("id" = "[TN.name]", "price" = "[price]", "time" = "[SQLtime()]"))
```
Comment (Jordie0608) [inline code/modules/research/rdconsole.dm]: I meant to add name as another argument not to replace the id =P
Also the var you want is `TN.display_name`

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

File: code/modules/research/rdconsole.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/rdconsole.dm]: "Oh you can webedit something so simple!" - Me, to me

**Review by Jordie0608 — APPROVED** [review]

File: code/modules/research/rdconsole.dm [hunk 2]
```diff
@@ -153,7 +153,7 @@ doesn't have toxins access.
 	if(stored_research.research_points >= price)
 		investigate_log("[key_name(user)] researched [id]([price]) on techweb id [stored_research.id].", INVESTIGATE_RESEARCH)
 		if(stored_research == SSresearch.science_tech)
-			SSblackbox.record_feedback("associative", "science_techweb_unlock", 1, list("id" = "[id]", "price" = "[price]", "time" = "[SQLtime()]"))
+			SSblackbox.record_feedback("associative", "science_techweb_unlock", 1, list("id" = "[id]", "name" = "[TN.display_name]", "price" = "[price]", "time" = "[SQLtime()]"))
```
Comment (Cyberboss) [inline code/modules/research/rdconsole.dm]: It doesn't need to be quoted

Comment (Cyberboss) [reply code/modules/research/rdconsole.dm]: Also unquote SQLtime while you're here please

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**